src/teachy_tv.c

```diff
@@ -0,0 +1,1571 @@
+#include "main.h"
+#include "global.h"
+#include "task.h"
+#include "menu.h"
+#include "palette.h"
+#include "sprite.h"
+#include "menu_helpers.h"
+#include "new_menu_helpers.h"
+#include "sound.h"
+#include "malloc.h"
+#include "scanline_effect.h"
+#include "bg.h"
+#include "gpu_regs.h"
+#include "decompress.h"
+#include "window.h"
+#include "list_menu.h"
+#include "item_menu.h"
+#include "item.h"
+#include "menu_indicators.h"
+#include "field_map_obj.h"
+#include "random.h"
+#include "text.h"
+#include "constants/songs.h"
+#include "constants/items.h"
+#include "event_data.h"
+#include "load_save.h"
+#include "battle_transition.h"
+#include "battle_2.h"
+#include "battle.h"
+#include "global.fieldmap.h"
+#include "teachy_tv.h"
+#include "help_system.h"
+#include "overworld.h"
+#include "unk_8159F40.h"
+
+struct TeachyTvCtrlBlk
+{
+    void (*callback)();
+    u8 mode;
+    u8 optionChosen;
+    u16 scrollOffset;
+    u16 selectedRow;
+    u16 filler;
+};
+
+struct TeachyTvBuf
+{
+    u32 state;
+    u8 buffer1[0x1000];
+    u8 buffer2[0x1000];
+    u8 buffer3[0x1000];
+    u8 buffer4[0x1000];
+    u8 var_4004;
+    u8 var_4005;
+    u8 var_4006;
+    u8 var_4007;
+};
+
+EWRAM_DATA struct TeachyTvCtrlBlk gTeachyTV_StaticResources = {0};
+EWRAM_DATA struct TeachyTvBuf * gUnknown_203F450;
+extern const struct ScrollIndicatorArrowPairTemplate gUnknown_8479380;
+extern const u8 gUnknown_841B83D;
```

Comment:
Please carefully review these declarations. A lot of them are arrays and of different type.